app/controllers/hosts_controller.rb

```diff
@@ -454,6 +454,18 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    res = {}
+    if @hosts.respond_to? :each
+      @hosts.each do |host|
```

Comment:
If you check the logs you will see there is an N+1 issue here, perhaps `@hosts.includes(:compute_resource).each` will fix it